src/validation.cpp

```diff
@@ -2083,7 +2084,7 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
     int64_t nSigOpsCost = 0;
     blockundo.vtxundo.reserve(block.vtx.size() - 1);
     std::vector<PrecomputedTransactionData> txdata;
-    txdata.reserve(block.vtx.size()); // Required so that pointers to individual PrecomputedTransactionData don't get invalidated
+    txdata.resize(block.vtx.size()); // Required so that pointers to individual PrecomputedTransactionData don't get invalidated
```

Comment:
nit, above could do `std::vector<PrecomputedTransactionData> txdata{block.vtx.size()};` and remove this,

Reply:
Done, and expanded the comment on why this vector is necessary.